hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveViewCommits.java

```diff
@@ -0,0 +1,474 @@
+  protected static final HiveMetastoreExtension HIVE_METASTORE_EXTENSION =
+      HiveMetastoreExtension.builder().withDatabase(DB_NAME).build();
+
+  private View view;
+
+  private static HiveCatalog catalog;
+
+  @BeforeAll
+  public static void initCatalog() {
+    catalog =
+        (HiveCatalog)
+            CatalogUtil.loadCatalog(
+                HiveCatalog.class.getName(),
+                CatalogUtil.ICEBERG_CATALOG_TYPE_HIVE,
+                ImmutableMap.of(
+                    CatalogProperties.CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS,
+                    String.valueOf(TimeUnit.SECONDS.toMillis(10))),
+                HIVE_METASTORE_EXTENSION.hiveConf());
+  }
+
+  @BeforeEach
+  public void createTestView() {
+    view =
+        catalog
+            .buildView(VIEW_IDENTIFIER)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(ns)
+            .withQuery("hive", "select * from ns.tbl")
+            .create();
+  }
+
+  @AfterEach
+  public void dropTestView() {
+    catalog.dropView(VIEW_IDENTIFIER);
+  }
+
+  @Test
+  public void testSuppressUnlockExceptions() {
+    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();
+    ViewMetadata metadataV1 = ops.current();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k1", "v1").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1);
+
+    HiveViewOperations spyOps = spy(ops);
+
+    AtomicReference<HiveLock> lockRef = new AtomicReference<>();
+
+    when(spyOps.lockObject(metadataV1.properties(), catalog.getConf(), catalog.name()))
+        .thenAnswer(
+            i -> {
+              HiveLock lock = (HiveLock) i.callRealMethod();
+              lockRef.set(lock);
+              return lock;
+            });
+
+    try {
+      spyOps.commit(metadataV2, metadataV1);
+      HiveLock spyLock = spy(lockRef.get());
+      doThrow(new RuntimeException()).when(spyLock).unlock();
+    } finally {
+      lockRef.get().unlock();
+    }
+
+    ops.refresh();
+
+    // the commit must succeed
+    assertThat(ops.current().properties()).hasSize(0);
+  }
+
+  /**
+   * Pretends we throw an error while persisting, and not found with check state, commit state
+   * should be treated as unknown, because in reality the persisting may still succeed, just not yet
+   * by the time of checking.
+   */
+  @Test
+  public void testThriftExceptionUnknownStateIfNotInHistoryFailureOnCommit()
+      throws TException, InterruptedException {
+    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k1", "v1").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1);
+
+    HiveViewOperations spyOps = spy(ops);
+
+    failCommitAndThrowException(spyOps);
+
```

Comment:
We are not checking 
```
assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
        .isInstanceOf(CommitStateUnknownException.class)
        .hasMessageStartingWith("Datacenter on fire");
 ```
 Like https://github.com/apache/iceberg/blob/main/hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java#L114C5-L116C55
 .
 There is a slight difference in the way we write newMetadataFile for both table and Views. 
 For table we have `newTable` check like 
 ```
 protected String writeNewMetadataIfRequired(boolean newTable, TableMetadata metadata) {
    return newTable && metadata.metadataFileLocation() != null
        ? metadata.metadataFileLocation()
        : writeNewMetadata(metadata, currentVersion() + 1);
  }
  ```

for View we dont have that check 
```
 protected String writeNewMetadataIfRequired(ViewMetadata metadata) {
    return null != metadata.metadataFileLocation()
        ? metadata.metadataFileLocation()
        : writeNewMetadata(metadata, version + 1);
  }
  ```
  Because of this , `spyOps.commit(metadataV2, metadataV1)` is successful for view.